src/clojure_ttt/ui.clj

```diff
@@ -55,17 +55,42 @@
        (some (partial = move))
        (boolean)))
 
-(defn input-move [board]
-  (let [s (input)]
-    (cond
+(defn input-move
+  ([board]
+   (input-move board 0))
+
+  ([board depth]
+   (cond
+     (= depth 0) (output "Please choose a move")
+     :else (output "Please choose a valid move"))
+
+   (let [s (input)]
+     (cond
+
+       (parseable-as-int? s)
+       (cond
+         (move-available? board (Integer/parseInt s))
+         (Integer/parseInt s)
 
-      (parseable-as-int? s)
-      (cond
-        (move-available? board (Integer/parseInt s))
-        (Integer/parseInt s)
+         :else
+         (recur board (inc depth)))
+
+       :else
+       (recur board (inc depth))))))
+
+(defn valid-player-type? [input-string]
+  (or (= input-string "h") (= input-string "c")))
+
+(defn input-player
+  ([]
+   (input-player 0))
 
-        :else
-        (recur board))
+  ([depth]
+   (cond
+     (= depth 0) (output "Choose player type: (h)uman or (c)omputer:")
```

Comment:
Similar comment to above - I think incrementing "depth" does not signal intent very clearly